forge/lib/tiler.py

```diff
@@ -54,6 +54,50 @@ def grid(bounds, zoomLevels, fullonly):
 tilecount = multiprocessing.Value('i', 0)
 skipcount = multiprocessing.Value('i', 0)
 
+visibility_timeout = 3600
+
+
+def createTileFromQueue(tq):
+    pid = os.getpid()
+    try:
+        (qName, t0, dbConfigFile) = tq
+        sqs = getSQS()
+        q = sqs.get_queue(qName)
+        geodetic = GlobalGeodetic(True)
+        # we do this as long as we are finding messages in the queue
+        while True:
+            parseOk = True
+            try:
+                # 20 is maximum wait time
+                m = q.read(visibility_timeout = visibility_timeout, wait_time_seconds = 20)
+                if m is None:
+                    logger.info('[%s] No more messages found. Closing process' % pid)
+                    break
+                body = m.get_body()
+                tiles = map(int, body.split(','))
+            except Exception as e:
+                    parseOk = False
+
+            if not parseOk or len(tiles) % 3 != 0:
+                logger.warning('[%s] Unparsable message received. Skipping...and removing message [%s]' % (pid, m.get_body()))
+                q.delete_message(m)
+                continue
```

Comment:
we don't know state of `tiles` beyond this point as it depends on where the exception occured when parseOk is False. It's to avoid the for loop, basically.